src/helper/securityQuestion.ts

```diff
@@ -0,0 +1,232 @@
+    await tkey._syncShareMetadata();
+
+    return factorKeyBN.toString("hex").padStart(64, "0");
+  }
+
+  async changeSecurityQuestion(params: changeSecurityQuestionParams) {
+    const { mpcCoreKit, newQuestion, newAnswer, answer } = params;
+    if (!newQuestion || !newAnswer || !answer) {
+      throw new Error("question and answer are required");
+    }
+    // Check for existing security question
+    const tkey = mpcCoreKit.tKey;
+
+    const domainKey = `${this.storeDomainName}:${params.mpcCoreKit.tKey.tssTag}`;
+    const storeDomain = tkey.metadata.getGeneralStoreDomain(domainKey) as StringifiedType;
+    if (!storeDomain || !storeDomain.question) {
+      throw new Error("Security question does not exists");
+    }
+
+    const store = TssSecurityQuestionStore.fromJSON(storeDomain);
+    let hash = keccak256(Buffer.from(answer, "utf8"));
+    hash = hash.startsWith("0x") ? hash.slice(2) : hash;
+    const decryptKey = Buffer.from(hash, "hex");
+
+    let factorKey;
+    try {
+      factorKey = await decrypt(decryptKey, store.associatedFactor);
+    } catch (error) {
+      throw new Error("Incorrect answer");
+    }
+
+    let newHash = keccak256(Buffer.from(newAnswer, "utf8"));
+    newHash = newHash.startsWith("0x") ? newHash.slice(2) : newHash;
+    const newAnswerBN = new BN(newHash, "hex");
+    const newEncryptKey = getPubKeyECC(newAnswerBN);
+    const newEncrypted = await encrypt(newEncryptKey, factorKey);
+
+    store.associatedFactor = newEncrypted;
+    store.question = newQuestion;
+    tkey.metadata.setGeneralStoreDomain(domainKey, store.toJSON());
+
+    // check for auto commit
+    await tkey._syncShareMetadata();
+  }
+
+  // Should we check with answer before deleting?
+  async deleteSecurityQuestion(mpcCoreKit: Web3AuthMPCCoreKit, deleteFactorKey = true) {
+    if (!mpcCoreKit.tKey) {
+      throw new Error("Tkey not initialized, call init first.");
+    }
+
+    const domainKey = `${this.storeDomainName}:${mpcCoreKit.tKey.tssTag}`;
+    const tkey = mpcCoreKit.tKey;
+    if (deleteFactorKey) {
+      const storeDomain = tkey.metadata.getGeneralStoreDomain(domainKey) as StringifiedType;
+      if (!storeDomain || !storeDomain.question) {
+        throw new Error("Security question does not exists");
+      }
+
+      const store = TssSecurityQuestionStore.fromJSON(storeDomain);
+      if (store.factorPublicKey) {
+        await mpcCoreKit.deleteFactor(tkeyPoint.fromCompressedPub(store.factorPublicKey));
+      }
+    }
+    const emptyStore = {};
+    tkey.metadata.setGeneralStoreDomain(domainKey, emptyStore);
+    // check for auto commit
+    await tkey._syncShareMetadata();
+  }
+
+  async recoverSecurityQuestionFactor(mpcCoreKit: Web3AuthMPCCoreKit, answer: string): Promise<string> {
+    if (!mpcCoreKit.tKey) {
+      throw new Error("Tkey not initialized, call init first.");
+    }
+    if (!answer) {
+      throw new Error("question and answer are required");
+    }
+    if (answer.length < 10) {
```

Comment:
i think we should not restrict the length, coz security question can be as simple as "Whats your first pet name?".

Reply:
That's how our security questions module is written as well in tKey, I think less than 10 might not be secure enough? @ieow do you know the reason behind it?

Reply:
Its UX vs security , application can enforce these checks if required, 10 chars are not ensuring any security anyways.

Reply:
done